src/Lecture2.hs

```diff
@@ -261,7 +314,23 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval variables expression = 
+  case expression of 
+    Var variable -> 
+       case lookup variable variables of
+          Just value -> Right value
+          Nothing -> Left (VariableNotFound variable)
+
+    Lit x -> 
+      Right x
+
+    Add a b -> evalSum (eval variables a) (eval variables b)
```

Comment:
You said you wrote an ugly solution 😅 
So I expected an ugly solution and not the perfect one 🤣